File: backend/src/main/java/br/com/aegro/production/services/FieldService.java
package br.com.aegro.production.services;

import br.com.aegro.production.domain.entities.Field;

import java.util.List;

public interface FieldService {

    List<Field> findByFarmId(String fieldId);

    Field findById(String fieldId);

    Field create(Field field);

    Field update(Field field);

    void deleteAll();

    void deleteById(String farmId);

    void deleteByFarmId(String farmId);

}

File: frontend/src/app/shared/services/field.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment as env } from '../../../environments/environment';

import { Field } from '../models';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class FieldService {

  private readonly URI: string = 'fields';

  constructor(
    private http: HttpClient
  ) { }

  findByFarmId(farmId: string): Observable<any> {
    return this.http.get(
      env.baseApiUrl + this.URI + "?farmId=" + farmId
    );
  }

  findById(fieldId: string): Observable<any> {
    return this.http.get(
      env.baseApiUrl + this.URI + "/" + fieldId
    );
  }

  insert(field: Field): Observable<any> {
    return this.http.post(
      env.baseApiUrl + this.URI,
      field
    );
  }

  update(field: Field): Observable<any> {
    return this.http.put(
      env.baseApiUrl + this.URI + "/" + field.id,
      field
    );
  }

  deleteById(fieldId: string): Observable<any> {
    return this.http.delete(
      env.baseApiUrl + this.URI + "/" + fieldId
    );
  }

}
